lib/generators/solidus_stripe/install/install_generator.rb

```diff
@@ -78,6 +78,18 @@ def install_solidus_starter_frontend_support
               template engine.root.join(path), path
             end
           end
+
+          storefront_layout_path = Rails.root.join("app/views/layouts/storefront.html.erb")
+
+          if storefront_layout_path.exist?
+            insert_into_file(
+              storefront_layout_path.to_s,
+              "\n    <script async src='https://js.stripe.com/v3/'></script>",
+              before: %r{\s*</head>},
+            )
+          end
+
+          run 'bin/importmap pin @stripe/stripe-js'
```

Comment:
That's not the case, although it might look like that at a first glance. For PCI compliance reasons the JS must be hosted on stripe.com and cannot be fully provided in the NPM package. In fact the package is nothing but a thin layer around loading the the JS from stripe.com.

The way it works is that if the `Stripe` constant is already available it will use that, and fallback to loading it on the fly if necessary, at the same time it provides the convenience of importing it as a module and having a Promise that can be waited on to ensure it's loaded.

Reply:
for the record: we discussed this live and removed the redundant global `<script>` tag in order to simplify the setup as the `stripe-js` import has a side-effect of adding it anyway as documented here: https://github.com/stripe/stripe-js#ensuring-stripejs-is-available-everywhere.